### PR TITLE
Handle case when device cannot be verified

### LINKED ISSUE #6466: `Verify this login` toast pops up when there is no way to do it (no other devices or 4S)
If you login on a new device and you don't have other active logins nor 4S, android will prompt you to verify but there is actually no way to do it.
image
image

For reference web is showing this:
image

## Short term fix:

Align with web. Show a popup stating that you have lost access to e2ee history and propose to reset cross signing and properly setup 4S.

## Long term fix.

This will need a bit more thinking.
But basically, why don't we just silently reset cross-signing keys?
Do we really need to do a full bootsrap (xsigning, backup + 4S?)?
Maybe this could be just an opportunity to explain the benefits of backup and 4S? And to explain why e2ee history is not accessible?

### PR DESCRIPTION
**Pull request by BillCarsonFr**
## Type of change

- [x] Bugfix

## Content

Fixes #6466

## Motivation and context

When there is no way to verify a new device, prompt and propose to reset verifications keys.

## Screenshots / GIFs

|Before|After|
|-|-|
|image|image|

This can be performed from settings also
image

A click on the popup will open the 4S setup bottomsheet

image

## Tests

- Step 1: Create an account then logout
- Step 2: Login this account again

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Review by Florian14 — COMMENTED** [review]
Not sure it has been changed in this PR but the "Reset everything" action sounds destructive, so I'd expect to show it in red, wdyt?

File: vector/src/main/java/im/vector/app/features/settings/devices/DeviceVerificationInfoBottomSheetController.kt [hunk 1]
```diff
@@ -141,19 +153,45 @@ class DeviceVerificationInfoBottomSheetController @Inject constructor(
...
+                        title(host.stringProvider.getString(R.string.secure_backup_reset_all))
+                        titleColor(host.colorProvider.getColorFromAttribute(R.attr.colorPrimary))
+                        iconRes(R.drawable.ic_arrow_right)
+                        iconColor(host.colorProvider.getColorFromAttribute(R.attr.colorPrimary))
+                        listener {
+                            host.callback?.onAction(DevicesAction.ResetSecurity)
+                        }
+                    }
+                }
+            } else if (!isMine) {
```
[the remark this answers is not in the document]
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/settings/devices/DeviceVerificationInfoBottomSheetController.kt:188]: yeah strange I used the editor split condition and it did that, will review the logic

**Comment by BillCarsonFr** [comment]
> Not sure it has been changed in this PR but the "Reset everything" action sounds destructive, so I'd expect to show it in red, wdyt?

In this case you have no recovery method, nor other device to verify against. So reset is your only choice, there is nothing to destruct and that's what you hsould do.

**Review by bmarty — APPROVED** [review]
LGTM

**Pull request merged**